[ButtonBase] Document how to use cursor not-allowed (#17778)
mui-org/material-ui: docs/src/pages/components/buttons/buttons.md

able to take advantage of it everywhere.

Here is an [integration example with react-router](/guides/composition/#button).

## Limitations

### Cursor not-allowed

> Material-UI's ButtonBase component set `pointer-events: none;` on disabled buttons.

Reviewer: ```suggestion
The ButtonBase component sets `pointer-events: none;` on disabled buttons.
```

> This tradeoff prevents the usage of a different disabled cursor.

Reviewer: ```suggestion
which prevents the appearance of a disabled cursor.
```

If you wish to use `not-allowed`, you have two options:

1. **CSS only**. You can remove the pointer events style on the disabled state of the `<button>` element:

```css
.MuiButtonBase-root:disabled {
  cursor: not-allowed;
  pointer-events: auto;
}
```

> However, you should add `pointer-events: none;` back when you need to display [tooltips on disabled elements](/components/tooltips/#disabled-elements) and the cursor won't change if you render something else than a button element, for instance, a link `<a>` element.

Reviewer: ```suggestion
However:

- You should add `pointer-events: none;` back when you need to display [tooltips on disabled elements](/components/tooltips/#disabled-elements).
- The cursor won't change if you render something other than a button element, for instance, a link `<a>` element.
```
